shell/platform/glfw/flutter_glfw.cc

```diff
@@ -690,6 +693,123 @@ static bool RunFlutterEngine(
   return true;
 }
 
+const char* GetLocaleStringFromEnvironmentVariables(const char* language,
+                                                    const char* lc_all,
+                                                    const char* lc_messages,
+                                                    const char* lang) {
+  const char* retval;
+  retval = language;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+  retval = lc_all;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+  retval = lc_messages;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+  retval = lang;
+  if ((retval != NULL) && (retval[0] != '\0')) {
+    return retval;
+  }
+
+  return NULL;
+}
+
+const char* GetLocaleStringFromEnvironment() {
+  return GetLocaleStringFromEnvironmentVariables(
+      getenv("LANGUAGE"), getenv("LC_ALL"), getenv("LC_MESSAGES"),
+      getenv("LANG"));
+}
+
+// Parse a locale into its components.
```

Comment:
Parses

Also, a declaration comment should be on the declaration in the header, rather than the definition.